### PR TITLE
Initial commit for mockcatalog module

### PR DESCRIPTION
**Pull request by fandinomat**
I have tested the mockcatalog module and it seems to work.
Comments are welcome!

### CODE REVIEW

**Comment by fandinomat** [comment]
@jrs65 @cahofer @gooddc

**Comment by fandinomat** [comment]
This pull request adds a new module: mockcatalog.py
It contains three pipeline tasks for generating mock quasar catalogs from a real catalog and a simulated h1 map. 
No other files are altered.

File: draco/synthesis/mockcatalog.py [hunk 1]
```diff
@@ -75,11 +75,11 @@
 from cora.signal import corr21cm
 from cora.util import units
 from caput import config
-from caput import mpiarray
+from caput import mpiarray, mpiutil
 from ..core import task, containers
 
 # For easy access to MPI namespace
-MPI_ = mpiarray.mpiutil.MPI
+MPI_ = mpiutil.MPI
```
Comment (jrs65) [inline draco/synthesis/mockcatalog.py]: The standard way to do this is to just do `from mpi4py import MPI`

**Comment by jrs65** [comment]
We should indeed. I am slowly working my way a backlog of PRs, so hopefully I'll get to this (and your thermal cal one) shortly.

**Comment by jrs65** [comment]
@fandinomat @sjforeman I think the plan we discussed for this branch was to close this PR and create a new PR that also includes the changes @sjforeman has made to it. I'm going to close this now, but feel free to reopen if you think we should deal with this differently.

**Pull request closed without merging**